### PR TITLE
bring back TermInterface

### PR DESCRIPTION
**Pull request by shashi**

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -528,28 +535,12 @@ end
...
-
-Create a term that is similar in type to `t`. Extending this function allows packages
-using their own expression types with SymbolicUtils to define how new terms should
-be created. Note that `similarterm` may return an object that has a
-different type than `t`, because `f` also influences the result.
-
-## Arguments
+function TermInterface.maketerm(::Type{<:BasicSymbolic}, head, args, type, metadata)
+    basicsymbolic(first(args), args[2:end], type, metadata)
+end
```
Comment (0x0f0f0f) [inline src/types.jl:540]: Please let's use the new one directly

Comment (0x0f0f0f) [reply src/types.jl:540]:
```
function TermInterface.maketerm(::Type{<:BasicSymbolic}, op, args, type=nothing, metadata=nothing)
    basicsymbolic(op, args, type, metadata)
end
```

File: Project.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name = "SymbolicUtils"
 uuid = "d1185830-fcd6-423d-90d6-eec64667417b"
 authors = ["Shashi Gowda"]
-version = "1.7.0"
+version = "1.8.0"
```
Comment (0x0f0f0f) [inline Project.toml:4]: Version 2.0?

File: src/types.jl (hunk 1 above)
Comment (nsajko) [reply src/types.jl:540]: xref https://github.com/JuliaSymbolics/TermInterface.jl/issues/33

**Pull request merged**